### DISCUSSION TITLE
Problem in opening word document with bad links

### DISCUSSION THREAD

**Discussion by ahmadjf93** [OfficeDev/Open-XML-SDK#1496] [Q&A]
I am unable to open word document using WordprocessingDocument.Open method because of bad links in the documents.
Please can you provide me a solution to bypass this problem. I am using openxml library in vb.net V2.0.5022.0

**Comment by mikeebowen** [comment]
Hi @ahmadjf93,

To help with this, we'll need some more information. What version of the SDK are you using? What error do you receive when you try to open the document? In what way are the links "bad"?

**Reply by ahmadjf93** [reply]
We are using SDK version 2.0 and the error received: "System.UriFormatException: 'Invalid URI: The hostname could not be parsed.'"
example of bad links an email "ajaafar@intellect" without .com

**Reply by mikeebowen** [reply]
If you update the SDK to v2.20.0 do you still see the issue?

**Reply by ahmadjf93** [reply]
Yes I tried this on my first step to solve the issue

**Reply by ahmadjf93** [reply]
@mikeebowen please can you provide me any solutions?

**Reply by ahmadjf93** [reply]
@mikeebowen please can you provide me any solution because we have this issue for long time and I think that you have a solution for this

**Reply by mikeebowen** [reply]
I will try and look into this this week. Another question, do the files in question open in Word?

**Reply by ahmadjf93** [reply]
Yes it opens normally

**Reply by ahmadjf93** [reply]
[TDCH-7840.docx](https://github.com/dotnet/Open-XML-SDK/files/12241591/TDCH-7840.docx)
this is the document

**Reply by ahmadjf93** [reply]
@mikeebowen Hello any news about this issue?

**Comment by mikeebowen** [comment]
Hi @ahmadjf93 , Sorry for the delay. I looked into this and it is a known issue with `System.IO.Packaging`. This issue has more details and discussion on the problem (https://github.com/dotnet/Open-XML-SDK/issues/38). There is a workaround that you can implement to handle malformed URIs. Eric White wrote an article about how to implement it on [his blog here](http://www.ericwhite.com/blog/handling-invalid-hyperlinks-openxmlpackageexception-in-the-open-xml-sdk/).
